### PR TITLE
Sensors update 11x

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

- allow comparison of sensors between different sensor units

## Related issues or pull requests

Please list issues or pull requests that the changes you propose are related to. It does not matter if they are still open and/or unmerged, any link is appreciated.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts [hunk 1]
```diff
@@ -103,29 +88,32 @@ export class HsSensorsUnitDialogComponent
 
   /**
    * @param interval - Clicked interval button
+   * @param generate - Controling wether to fetch observations and generate charts as well. Not necessary on init
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts:91]:
```suggestion
   * @param generate - Controlling whether to fetch observations and generate charts as well. Not necessary on init
```

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts [hunk 2]
```diff
@@ -103,29 +88,32 @@ export class HsSensorsUnitDialogComponent
...
-        interval,
-        this.data.app
-      )
-      .then((_) => {
-        this.hsSensorsUnitDialogService.createChart(
-          this.hsSensorsUnitDialogService.get(this.data.app).unit,
+    if (generate) {
+      this.hsSensorsUnitDialogService
+        .getObservationHistory(
+          this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //ALWAYS JUST ONE?
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts:106]:
```suggestion
          this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //FIXME: ALWAYS JUST ONE?
```

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts [hunk 3]
```diff
@@ -103,29 +88,32 @@ export class HsSensorsUnitDialogComponent
...
+          interval,
           this.data.app
-        );
-      });
+        )
+        .then((_) => {
+          this.hsSensorsUnitDialogService.createChart(
+            this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //ALWAYS JUST ONE?
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts:112]:
```suggestion
            this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //FIXME: ALWAYS JUST ONE?
```

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts [hunk 4]
```diff
@@ -136,13 +124,13 @@ export class HsSensorsUnitDialogComponent
       this.customInterval;
     this.hsSensorsUnitDialogService
       .getObservationHistory(
-        this.hsSensorsUnitDialogService.get(this.data.app).unit,
+        this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //ALWAYS JUST ONE?
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts:127]:
```suggestion
        this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //FIXME: ALWAYS JUST ONE?
```

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts [hunk 5]
```diff
@@ -136,13 +124,13 @@ export class HsSensorsUnitDialogComponent
...
         this.customInterval,
         this.data.app
       )
       .then((_) =>
         this.hsSensorsUnitDialogService.createChart(
-          this.hsSensorsUnitDialogService.get(this.data.app).unit,
+          this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //ALWAYS JUST ONE?
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts:133]:
```suggestion
          this.hsSensorsUnitDialogService.get(this.data.app).unit[0], //FIXME: ALWAYS JUST ONE?
```

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-list-item.component.ts [hunk 6]
```diff
@@ -57,16 +58,29 @@ export class HsSensorsUnitListItemComponent {
    * displaying charts or reopen already existing one.
    */
   sensorToggleSelected(sensor): void {
-    this.hsSensorsUnitDialogService.get(this.app).unit = this.unit;
     sensor.checked = !sensor.checked;
+    const unitDialogServiceRef = this.hsSensorsUnitDialogService.get(this.app);
+    if (unitDialogServiceRef.comparisonAllowed) {
+      //If the opened sensor belongs to unit thats not included add it
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors-unit-list-item.component.ts:64]:
```suggestion
      //If the opened sensor belongs to unit that's not included add it
```
missing end of the sentence

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-list-item.component.ts [hunk 7]
```diff
@@ -57,16 +58,29 @@ export class HsSensorsUnitListItemComponent {
...
+      //NOTE: Might not even be possible asi checkbxes are available only after unit is selected
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors-unit-list-item.component.ts:65]:
```suggestion
      //NOTE: Might not even be possible as checkboxes are available only after unit is selected
```

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-list-item.component.ts [hunk 8]
```diff
@@ -57,16 +58,29 @@ export class HsSensorsUnitListItemComponent {
...
+    }
+
     this.hsSensorsUnitDialogService.toggleSensor(sensor, this.app);
-    this.generateDialog();
+    this.generateDialog(!unitDialogServiceRef.comparisonAllowed);
   }
 
   /**
    * Display sensors unit dialog
+   * @single Controls wether only one unit is supposed to be selected
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors-unit-list-item.component.ts:81]:
```suggestion
   * @single Controls whether only one unit is supposed to be selected
```

File: projects/hslayers-sensors/src/components/sensors/sensors.service.ts [hunk 9]
```diff
@@ -407,14 +489,19 @@ export class HsSensorsService {
...
             value: sv.observedValue,
             timestamp: dayjs(sv.timeStamp).format('DD.MM.YYYY HH:mm'),
           };
         });
         appRef.units.forEach((unit: HsSensorUnit) => {
           unit.sensors.forEach((sensor) => {
-            const reading = sensorValues[unit.unit_id + sensor.sensor_id];
+            /**
+             * NOTE:
+             *  In order to distinguish between sensors in differnet units sensor_id
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/sensors.service.ts:501]:
```suggestion
             *  In order to distinguish between sensors in different units sensor_id
```

File: projects/hslayers-sensors/src/components/sensors/unit-dialog.service.ts [hunk 10]
```diff
@@ -35,6 +39,12 @@ class SensorsUnitDialogServiceParams {
 
   timeFormat: 'HH:mm:ss' | 'HH:mm:ssZ';
   useTimeZone = new BehaviorSubject<boolean>(false);
+
+  /**
+   * Controls wether its possible to compare sensors inbetween
```
Comment (jmacura) [inline projects/hslayers-sensors/src/components/sensors/unit-dialog.service.ts:44]:
```suggestion
   * Controls whether it's possible to compare sensors in between
```

File: projects/hslayers/src/assets/locales/cs.json [hunk 11]
```diff
@@ -866,7 +866,10 @@
     "oneHour": "1 hodina",
     "phenomena": "Jevy",
     "sensors": "Senzory",
-    "sensorUnit": "Senzorová jednotka"
+    "sensorUnit": "Senzorová jednotka",
+    "comparison": "Porovnat senzory mezi jednotkamiPorovnať senzory medzi jednotkami",
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:870]:
```suggestion
    "comparison": "Porovnat senzory mezi jednotkami",
```

**Review by jmacura — COMMENTED** [review]
I have only a shallow understanding about what is this change about. Hence I am adding some formal review comments.

File: projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts (hunk 2 above)
Comment (FilipLeitner) [reply projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts]: Thanks for pointing these out, forgot to get back and fix it. Its now done in [6c8dfc8](https://github.com/hslayers/hslayers-ng/pull/4100/commits/6c8dfc86d7c3ff075f58f331acd3a605cedee086) and [32811e8](https://github.com/hslayers/hslayers-ng/pull/4100/commits/32811e81dde8c5c5931a3705c58f79a8d5e1d4b0)

Comment (jmacura) [reply projects/hslayers-sensors/src/components/sensors/sensors-unit-dialog.component.ts]: Great, thanks!

**Pull request merged**